tests/UnitTests/Foundation/NSObject_NSKeyValueOrderedSetAdaptersTests.mm

```diff
@@ -0,0 +1,265 @@
+    fakeMutableInserted = YES;
+}
+
+- (void)removeObjectFromFakeMutableCollectionAtIndex:(NSUInteger)index {
+    fakeMutableRemoved = YES;
+}
+
+- (void)insertFakeMutableCollection:(NSArray*)objects atIndexes:(NSIndexSet*)indexes {
+    fakeMutableInsertedAtIndexes = YES;
+}
+
+- (void)removeFakeMutableCollectionAtIndexes:(NSIndexSet*)indexes {
+    fakeMutableRemovedAtIndexes = YES;
+}
+
+- (BOOL)fakeMutableInserted {
+    return fakeMutableInserted;
+}
+
+- (BOOL)fakeMutableRemoved {
+    return fakeMutableRemoved;
+}
+
+- (BOOL)fakeMutableInsertedAtIndexes {
+    return fakeMutableInsertedAtIndexes;
+}
+
+- (BOOL)fakeMutableRemovedAtIndexes {
+    return fakeMutableRemovedAtIndexes;
+}
+
+- (NSOrderedSet*)immutableWithAccessors {
+    return [[_internal_immutableWithAccessors retain] autorelease];
+}
+
+- (void)setImmutableWithAccessors:(NSOrderedSet*)value {
+    [_internal_immutableWithAccessors release];
+    _internal_immutableWithAccessors = [value copy];
+    immutableWithAccessorsOrderedSet = YES;
+}
+
+- (BOOL)immutableWithAccessorsOrderedSet {
+    return immutableWithAccessorsOrderedSet;
+}
+
+- (id)valueForKey:(NSString*)key {
+    if ([@"immutableThroughKVC" isEqual:key]) {
+        return [[_internal_immutableThroughKVC retain] autorelease];
+    }
+    return [super valueForKey:key];
+}
+
+- (void)setValue:(id)value forKey:(NSString*)key {
+    if ([@"immutableThroughKVC" isEqual:key]) {
+        [_internal_immutableThroughKVC release];
+        _internal_immutableThroughKVC = [value copy];
+        immutableThroughKVCOrderedSet = YES;
+        return;
+    }
+    return [super setValue:value forKey:key];
+}
+
+- (BOOL)immutableThroughKVCOrderedSet {
+    return immutableThroughKVCOrderedSet;
+}
+@end
+
+TEST(NSObject, KVCOrderedSetMutableAdapters_CompliantIndexedAccessors) {
+    TestOrderedSetAdapterObject* testObject = [[[TestOrderedSetAdapterObject alloc] init] autorelease];
+    NSMutableOrderedSet* fakeMutableCollection = [testObject mutableOrderedSetValueForKey:@"fakeMutableCollection"];
+    EXPECT_TRUE([fakeMutableCollection isKindOfClass:[NSMutableOrderedSet class]]);
+
+    EXPECT_EQ(10, [fakeMutableCollection count]);
+    EXPECT_OBJCEQ(@5, [fakeMutableCollection objectAtIndex:5]);
+}
+
+TEST(NSObject, KVCOrderedSetMutableAdapters_InsertObject) {
+    TestOrderedSetAdapterObject* testObject = [[[TestOrderedSetAdapterObject alloc] init] autorelease];
+    NSMutableOrderedSet* fakeCollection = [testObject mutableOrderedSetValueForKey:@"fakeMutableCollection"];
+    EXPECT_NO_THROW([fakeCollection insertObject:@"TEST" atIndex:0]);
+    EXPECT_TRUE([testObject fakeMutableInserted]);
+}
+
+TEST(NSObject, KVCOrderedSetMutableAdapters_RemoveObject) {
+    TestOrderedSetAdapterObject* testObject = [[[TestOrderedSetAdapterObject alloc] init] autorelease];
+    id fakeCollection = [testObject mutableOrderedSetValueForKey:@"fakeMutableCollection"];
```

Comment:
unfortunately these cannot be NSMutableOrderedSet* because they're currently STUBs and the compiler thinks it wrong.